Wrappers/WebElement.cs

```diff
@@ -0,0 +1,130 @@
+﻿using OpenQA.Selenium;
+using SeleniumExtras.WaitHelpers;
+using System.Collections.ObjectModel;
+using System.Drawing;
+using Utilities;
+
+namespace Wrappers
+{
+    public class WebPageElement : IWebElement
+
+    {
+        protected IWebDriver Driver;
+        protected IWebElement Element;
+
+        public WebPageElement(IWebElement element)
+        {
+            Element = element;
+        }
+
+        public WebPageElement(IWebDriver driver, By locator)
```

Comment:
public WebPageElement(**IWebDriver driver**, By locator)

I don't like this. It force us to pass `driver` every time we call constructor. Where are you going to take this driver? Probably from some `BrowserFactory` or like this. So your code will be:

```
var driver = BrowserFactory.Driver;
var btn = new WebPageElement(driver, By.XPath("xpath"));
```

Instead, you can access driver directly from your WebPageElement class. Change Driver property to this:
```
    protected IWebDriver Driver => BrowserFactory.Driver;
```
And now your constructor will be like this:
```
public WebPageElement(By locator)
{
    Element = Driver.FindElements(locator).FirstOrDefault();
}
```

Reply:
Fixed